src/ast/irbuilderbpf.cpp

```diff
@@ -265,7 +267,7 @@ void IRBuilderBPF::CreateProbeRead(AllocaInst *dst, size_t size, Value *src)
   // Return: 0 on success or negative error
   FunctionType *proberead_func_type = FunctionType::get(
       getInt64Ty(),
-      {getInt8PtrTy(), getInt64Ty(), getInt8PtrTy()},
+      {dst->getType(), getInt64Ty(), src->getType()},
```

Comment:
Isn't the second arg a u32? Not a int64?

https://github.com/torvalds/linux/blob/master/include/uapi/linux/bpf.h#L555